app/models/task.rb

```diff
@@ -264,6 +264,44 @@ def do_hold(last_check)
     end
   end
 
+  # Returns if task is the root task or not
+  def root
+    return true if self.id == schedule.root_task_id
+    false
+  end
+
+
+  # Returns time (in milliseconds) when task begins
+  def trigger_start
+    return 0 if self.root?
+
+    case self.trigger
```

Comment:
Rails enums give you functions that end in ? for confirming whether the enum is in a given state. So, instead of comparing trigger to some strings, it should look like:
`ruby
case
  when on_started?
    # stuff
  when on_done?
    # other stuff
  else
end

`